File: tools/config-tui/src/lib/file-writer.ts
import { writeFileSync } from "node:fs";
import { resolve } from "node:path";
import { configRoot } from "./validator.js";

export function saveAppConfig(data: Record<string, unknown>): void {
  const json = JSON.stringify(data, null, 2) + "\n";
  writeFileSync(resolve(configRoot, "app-config.json"), json, "utf-8");
}

export function saveThemeConfig(data: Record<string, unknown>): void {
  const json = JSON.stringify(data, null, 2) + "\n";
  writeFileSync(resolve(configRoot, "theme.json"), json, "utf-8");
}

File: tools/config-tui-py/vonage_config_tui/file_writer.py
"""Writes app-config.json / theme.json back to disk"""

from __future__ import annotations

import json
from typing import Any

from .validator import CONFIG_ROOT


def save_app_config(data: dict[str, Any]) -> None:
    (CONFIG_ROOT / "app-config.json").write_text(
        json.dumps(data, indent=2) + "\n", encoding="utf-8"
    )


def save_theme_config(data: dict[str, Any]) -> None:
    (CONFIG_ROOT / "theme.json").write_text(
        json.dumps(data, indent=2) + "\n", encoding="utf-8"
    )
